File: src/Plotly.NET/Marker.fs
namespace Plotly.NET

open DynamicObj
open System

/// Marker type inherits from dynamic object
type Marker () =
    inherit DynamicObj ()

    /// Initialized Marker object
    static member init
        (   
            ?Size: int,
            ?Opacity: float,
            ?Color: string,
            ?Symbol: StyleParam.Symbol,
            ?MultiSizes: seq<#IConvertible>,
            ?Line: Line,
            ?Colorbar: Colorbar,
            ?Colorscale : StyleParam.Colorscale,
            ?Colors: seq<string>,
            ?OutlierColor:string,
                            
            ?Maxdisplayed: int,
            ?Sizeref: float,
            ?Sizemin: float,
            ?Sizemode: StyleParam.SizeMode,
            ?Cauto: bool,
            ?Cmax: float,
            ?Cmin: float,
            ?Cmid: float,
            ?Autocolorscale: bool,
            ?Reversescale: bool,
            ?Showscale: bool

        ) =
        Marker () 
        |> Marker.style
            (
                ?Size          = Size           ,
                ?Color         = Color          ,
                ?Symbol        = Symbol         ,
                ?Opacity       = Opacity        ,
                ?MultiSizes    = MultiSizes     ,
                ?Line          = Line           ,
                ?Colorbar      = Colorbar       ,
                ?Colorscale    = Colorscale     ,
                ?Colors        = Colors         ,
                ?OutlierColor  = OutlierColor   ,
                                        
                ?Maxdisplayed  = Maxdisplayed   ,
                ?Sizeref       = Sizeref        ,
                ?Sizemin       = Sizemin        ,
                ?Sizemode      = Sizemode       ,
                ?Cauto         = Cauto          ,
                ?Cmax          = Cmax           ,
                ?Cmin          = Cmin           ,
                ?Autocolorscale= Autocolorscale ,
                ?Reversescale  = Reversescale   ,
                ?Showscale     = Showscale      

            )

    // Applies the styles to Marker()
    static member style
        (   
            ?Size: int,
            ?Opacity: float,
            ?Color: string,
            ?Symbol: StyleParam.Symbol,
            ?MultiSizes: seq<#IConvertible>,
            ?Line: Line,
            ?Colorbar: Colorbar,
            ?Colorscale : StyleParam.Colorscale,
            ?Colors: seq<string>,
            ?OutlierColor:string,
                            
            ?Maxdisplayed: int,
            ?Sizeref: float,
            ?Sizemin: float,
            ?Sizemode: StyleParam.SizeMode,
            ?Cauto: bool,
            ?Cmax: float,
            ?Cmin: float,
            ?Cmid: float,
            ?Autocolorscale: bool,
            ?Reversescale: bool,
            ?Showscale: bool
        ) =
            (fun (marker: Marker) -> 
                Size           |> DynObj.setValueOpt marker "size" 
                Color          |> DynObj.setValueOpt marker "color"
                Symbol         |> DynObj.setValueOpt marker "symbol"
                Opacity        |> DynObj.setValueOpt marker "opacity"
                MultiSizes     |> DynObj.setValueOpt marker "size"
                Line           |> DynObj.setValueOpt marker "line"        
                Colorbar       |> DynObj.setValueOpt marker "colorbar"       
                Colorscale     |> DynObj.setValueOptBy marker "colorscale" StyleParam.Colorscale.convert
                Colors         |> DynObj.setValueOpt marker "colors"     
                OutlierColor   |> DynObj.setValueOpt marker "outliercolor"     
                                                
                Maxdisplayed   |> DynObj.setValueOpt marker "maxdisplayed"   
                Sizeref        |> DynObj.setValueOpt marker "sizeref"        
                Sizemin        |> DynObj.setValueOpt marker "sizemin"        
                Sizemode       |> DynObj.setValueOpt marker "sizemode"            
                Cauto          |> DynObj.setValueOpt marker "cauto"          
                Cmax           |> DynObj.setValueOpt marker "cmax"           
                Cmin           |> DynObj.setValueOpt marker "cmin"           
                Autocolorscale |> DynObj.setValueOpt marker "autocolorscale" 
                Reversescale   |> DynObj.setValueOpt marker "reversescale"   
                Showscale      |> DynObj.setValueOpt marker "showscale"      

                marker
            )

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Marker.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.Marker Marker
        (

            Optional<bool> AutoColorScale = default,
            Optional<bool> CAuto = default,
            Optional<double> CMax = default,
            Optional<double> CMid = default,
            Optional<double> CMin = default,
            Optional<Color> Color = default,
            Optional<IEnumerable<Color>> Colors = default,
            Optional<StyleParam.SubPlotId> ColorAxis = default,
            Optional<ColorBar> ColorBar = default,
            Optional<StyleParam.Colorscale> Colorscale = default,
            Optional<Gradient> Gradient = default,
            Optional<Line> Outline = default,
            Optional<int> MaxDisplayed = default,
            Optional<double> Opacity = default,
            Optional<IEnumerable<double>> MultiOpacity = default,
            Optional<Pattern> Pattern = default,
            Optional<bool> ReverseScale = default,
            Optional<bool> ShowScale = default,
            Optional<int> Size = default,
            Optional<IEnumerable<int>> MultiSize = default,
            Optional<int> SizeMin = default,
            Optional<StyleParam.MarkerSizeMode> SizeMode = default,
            Optional<int> SizeRef = default,
            Optional<StyleParam.MarkerSymbol> Symbol = default,
            Optional<IEnumerable<StyleParam.MarkerSymbol>> MultiSymbol = default,
            Optional<StyleParam.MarkerSymbol3D> Symbol3D = default,
            Optional<IEnumerable<StyleParam.MarkerSymbol3D>> MultiSymbol3D = default,
            Optional<Color> OutlierColor = default,
            Optional<int> OutlierWidth = default
        ) => 
            Plotly.NET.TraceObjects.Marker.init(

                AutoColorScale: AutoColorScale.ToOption(),
                CAuto: CAuto.ToOption(),
                CMax: CMax.ToOption(),
                CMid: CMid.ToOption(),
                CMin: CMin.ToOption(),
                Color: Color.ToOption(),
                Colors: Colors.ToOption(),
                ColorAxis: ColorAxis.ToOption(),
                ColorBar: ColorBar.ToOption(),
                Colorscale: Colorscale.ToOption(),
                Gradient: Gradient.ToOption(),
                Outline: Outline.ToOption(),
                MaxDisplayed: MaxDisplayed.ToOption(),
                Opacity: Opacity.ToOption(),
                MultiOpacity: MultiOpacity.ToOption(),
                Pattern: Pattern.ToOption(),
                ReverseScale: ReverseScale.ToOption(),
                ShowScale: ShowScale.ToOption(),
                Size: Size.ToOption(),
                MultiSize: MultiSize.ToOption(),
                SizeMin: SizeMin.ToOption(),
                SizeMode: SizeMode.ToOption(),
                SizeRef: SizeRef.ToOption(),
                Symbol: Symbol.ToOption(),
                MultiSymbol: MultiSymbol.ToOption(),
                Symbol3D: Symbol3D.ToOption(),
                MultiSymbol3D: MultiSymbol3D.ToOption(),
                OutlierColor: OutlierColor.ToOption(),
                OutlierWidth: OutlierWidth.ToOption()
            );
}

